src/redux/selectors/index.js

```diff
@@ -194,8 +178,94 @@ export const dateRangeForAllRecordsSelector = createSelector(
   },
 );
 
-export const dateRangeForFilteredRecordsSelector = createSelector(
-  [filteredRecordsSelector],
+// returns Array of (picked fields from) _all_ records, flagged with responsiveness to each filter:
+const allRecordsWithFilterResponseSelector = createSelector(
+  [allValidRecordsSortedByDateSelector, activeCollectionSelector, dateRangeForAllRecordsSelector],
+  (items, activeCollection, dateRangeForAllRecords) => {
+    const { minimumDate, maximumDate } = dateRangeForAllRecords;
+    const {
+      resourceTypeFilters,
+      showCollectionOnly,
+      showMarkedOnly,
+      records,
+      dateRangeFilter: { dateRangeStart = minimumDate, dateRangeEnd = maximumDate },
+    } = activeCollection;
+
+    return items.map(({
+      id, type, subType, timelineDate,
+    }) => ({
+      id,
+      type,
+      subType,
+      timelineDate,
+      passesFilters: {
+        type: resourceTypeFilters[type],
+        date: dateRangeStart && dateRangeEnd && isWithinInterval(
+          timelineDate,
+          {
+            start: dateRangeStart,
+            end: dateRangeEnd,
+          },
+        ),
+        inCollection: records[id]?.saved,
+        showCollectionOnly: !showCollectionOnly || (showCollectionOnly && records[id]?.saved),
+        isHighlighted: records[id]?.highlight,
+        showHighlightedOnly: !showMarkedOnly || (showMarkedOnly && records[id]?.highlight),
+      },
+    }));
+  },
+);
+
+// is there a record with all filters applied, other than showCollectionOnly?
+export const hasAnyCollectionRecordInScope = createSelector(
+  [allRecordsWithFilterResponseSelector],
+  (items) => !!items.find(({
+    passesFilters: {
+      type,
+      date,
+      inCollection,
+      // showCollectionOnly,
+      // isHighlighted,
+      showHighlightedOnly,
+    },
+  }) => type && date && showHighlightedOnly && inCollection),
```

Comment:
shouldn't `hasAnyCollectionRecordInScope` ignore `showHighlightedOnly` attribute?

Reply:
it has to pass all filters other than itself